File: egg.js
require("dotenv"); // Load env configuration

const Discord = require('discord.js');
const client = new Discord.Client();

let eggCount = 0;

client.on('ready', () => {
	console.log(`Logged in as ${client.user.tag}!`);
});

client.on('message', msg => {
	if(msg.content.toLowerCase().includes("egg")){
		msg.react('🥚');
		eggCount++;
		client.user.setPresence({
			activity: { 
				name: `${eggCount} 🥚`
			},
			status: 'online'
		})
	}
});
client.on('message', msg => {
if(msg.channel.fetchMessage(261323096617779201)){
msg.delete();
}};
client.login(process.env.EGG_BOT_TOKEN);

File: egg.nim
import dimscord, asyncdispatch, times, options, os, parseutils, strutils

let discord = newDiscordClient(getEnv("EGG_BOT_TOKEN"))
var eggs = 0

if fileExists("eggs.txt"):
  discard readFile("eggs.txt").parseInt(eggs)

proc onReady(s: Shard, r: Ready) {.event(discord).} =
  echo "Ready as " & $r.user
  await s.updateStatus(activity = ActivityStatus(
    name: $eggs & " 🥚",
    kind: atPlaying
  ).some, status = "online")

proc messageCreate(s: Shard, m: Message) {.event(discord).} =
  if not m.author.bot and m.content.toLowerAscii.contains("egg"):
    eggs.inc

    try:
      await discord.api.addMessageReaction(m.channelId, m.id, "🥚")
      await s.updateStatus(activity = ActivityStatus(
        name: $eggs & " 🥚",
        kind: atPlaying
      ).some, status = "online")
      writeFile("eggs.txt", $eggs)
    except:
      echo getCurrentExceptionMsg()

waitFor discord.startSession()